datasets/diseases/scripts/gold_standard.py

```diff
@@ -0,0 +1,111 @@
+import pandas as pd
+import requests
+import os
+from pathlib import Path
+import pickle
+
+# https://stackoverflow.com/a/5137509/7589775
+dir_path = os.path.dirname(os.path.realpath(__file__))
+
+diseases_path = Path(dir_path, '..')
+
+def main():
+    # Get our data from `fetch.py`
+    text_mining = pd.read_csv(diseases_path / "raw" / "human_disease_textmining_filtered.tsv", sep="\t")
+    knowledge = pd.read_csv(diseases_path / "raw" / "human_disease_knowledge_filtered.tsv", sep="\t")
+    # and correctly map their columns
+    text_mining.columns = ["geneID", "geneName", "diseaseID", "diseaseName", "zScore", "confidenceScore", "sourceUrl"]
+    knowledge.columns = ["geneID", "geneName", "diseaseID", "diseaseName", "sourceDB", "evidenceType", "confidenceScore"]
+
+    # The DISEASES data is in the ENSP namespace, but we want to work in ENSG.
+    knowledge_mapped = gprofiler_convert(list(text_mining["geneID"]), "ENSG", text_mining)
+    text_mining_mapped = gprofiler_convert(list(knowledge["geneID"]), "ENSG", knowledge)
+
+    inner = text_mining_mapped.merge(knowledge_mapped, on=["ENSG", "diseaseID"], how="inner")
+    inner["confidenceScore"] = inner.apply(lambda x: max(x.confidenceScore_x, x.confidenceScore_y), axis=1)
+    inner = inner.rename(columns={"ENSP_x": "ENSP", "geneName_x": "geneName", "diseaseName_x": "diseaseName", "geneID_x": "geneID"})
+    inner = inner[["ENSG", "ENSP", "geneName", "diseaseID", "diseaseName", "confidenceScore"]]
+
+    txt_only = text_mining_mapped.merge(knowledge_mapped, on=["ENSG", "diseaseID"], how="left")
+    txt_only = txt_only[txt_only["confidenceScore_y"].isna()]
+    txt_only = txt_only.rename(
+        columns={"confidenceScore_x": "confidenceScore", "ENSP_x": "ENSP", "geneName_x": "geneName", "diseaseName_x": "diseaseName"}
+    )
+    txt_only = txt_only[["ENSG", "ENSP", "geneName", "diseaseID", "diseaseName", "confidenceScore"]]
+
+    kn_only = text_mining_mapped.merge(knowledge_mapped, on=["ENSG", "diseaseID"], how="right")
+    kn_only = kn_only[kn_only["confidenceScore_x"].isna()]
+    kn_only = kn_only.rename(
+        columns={"confidenceScore_y": "confidenceScore", "ENSP_y": "ENSP", "geneName_y": "geneName", "diseaseName_y": "diseaseName"}
+    )
+    kn_only = kn_only[["ENSG", "ENSP", "geneName", "diseaseID", "diseaseName", "confidenceScore"]]
+
+    df_list = [inner, txt_only, kn_only]
+    GS_ids = pd.concat(df_list)
+
+    # GS_group = GS_ids.groupby('diseaseName')
+    # GS_dict = {k:v for k,v in GS_group}
+    # GS_count = {x:len(GS_dict[x]) for x in GS_dict.keys()}
+    # print('count quantiles: ',stats.quantiles(GS_count.values()))
+    # print('score quantiles: ',stats.quantiles(GS_ids['confidenceScore']))
```

Comment:
(do we need this for anything? This is a leftover of the jupyter notebook, and we can just let it decay in git history instead.)